doc/sphinx/source/conf.py

```diff
@@ -46,6 +47,13 @@
 if rtd_version not in ["latest", "stable", "doc"]:
     rtd_version = "latest"
 
+
+if re.fullmatch("v\d+\.\d+\.\d+", __version__):
+    baseurl = f"https://github.com/ESMValGroup/ESMValTool/tree/v{__version__}"
+else:
+    baseurl = f"https://github.com/ESMValGroup/ESMValTool/tree/main"
```

Comment:
for a pointer to a branch you can use this:

```python
import git


def get_current_branch_name(repo_path):
    """
    Gets the name of the active Git branch as a string.
    """
    repo = git.Repo(repo_path)
    branch = repo.active_branch
    return branch.name

branch = get_current_branch_name(".")
baseurl = f"https://github.com/ESMValGroup/ESMValTool/tree/{branch}"
```

this would need `gitpython` from conda-forge :beer:

Reply:
Thanks @valeriupredoi ! Very elegent.

Recording our discussion for my own memory - it would additionally be good to test that `branch` is a valid string, with `if(branch)`

Reply:
could do `if branch:`` or not even at all since the thing is 100% running inside the git repo where the branch will always be returned